src/server/api/routers/posts.ts

```diff
@@ -0,0 +1,9 @@
+import { z } from "zod";
+
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+
+export const postRouter = createTRPCRouter({
+  getAll: publicProcedure.query(({ ctx }) => {
```

Comment:
```suggestion
  getAll: publicProcedure.query( async ({ ctx }) => {
```

こちらもしかしたら、asyncを使用する方がよいかもしれません！
DBとの通信を行う関数になりそうなので、
意図がなければ、async/awaitを使用した方が型の利用がしやすいと思います！

よければこちらご参考にしていただけると良いかと思います！
[typescript async](https://typescriptbook.jp/reference/asynchronous/async)